batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,379 @@
+#include "batch_job.h"
+#include "batch_job_internal.h"
+#include "debug.h"
+#include "process.h"
+#include "macros.h"
+#include "stringtools.h"
+#include "path.h"
+#include "xxmalloc.h"
+#include "jx.h"
+#include "jx_print.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+
+#define FILE_TASK_INFO "task_info"
+#define FILE_TASK_STATE "task_state"
+
+#define NUM_OF_TASKS 4096
+
+static int counter = 0;
+static int finished_tasks[NUM_OF_TASKS];
+static int num_finished_tasks = 0;
+static time_t old_time;
+static int is_first_time = 1;
+
+// mesos task struct
+typedef struct mesos_task{
```

Comment:
To be consistent with the rest of the cctools, please name the structure and function like so:

```
struct mesos_task {};
struct mesos_task * mesos_task_create( . . .);
void mesos_task_delete( struct mesos_task * );
```

You might even move the data type into `mesos_task.[ch]` to keep it separate.